compiler-rt/lib/sanitizer_common/sanitizer_linux.cpp

```diff
@@ -1214,6 +1220,16 @@ uptr GetPageSize() {
   CHECK_EQ(rv, 0);
   return (uptr)pz;
 #    elif SANITIZER_USE_GETAUXVAL
+#      if SANITIZER_ANDROID && __ANDROID_API__ < 35
+  // The 16 KB page size was introduced in Android 15 (API level 35), while
```

Comment:
But not unconditionally. This is wrong on ~every device. Can you not use `getpagesize()`? Or the `sysconf(_SC_PAGESIZE)` that's the fallback here?

Reply:
```fish
❯ echo '#include <sys/sysconf.h>'\n'static void preinit_fn() {sysconf(_SC_PAGESIZE);}; void (*preinit[])()__attribute((section(".preinit_array")))={&preinit_fn};int main(){}' | cc -x c -;./a.out
fish: Job 1, './a.out' terminated by signal SIGSEGV (Address boundary error)
```

```fish
❯ echo '#include <unistd.h>'\n'static int page_size; static void preinit_fn() {page_size = getpagesize();}; void (*preinit[])()__attribute((section(".preinit_array")))={&preinit_fn};int main(){__builtin_printf("%i", page_size);}' | cc -x c -;./a.out
4096⏎    
```
`getpagesize` does work. Sorry, I never tried it before.

so Something like this?
```c++
#      if SANITIZER_ANDROID
  return getpagesize();
#else 
  return getauxval(AT_PAGESZ);
#      endif
```

Reply:
we can't use getpagesize for the same reason as above. it is correct that with old API levels, pagesize was a 4k constant.

Reply:
I think `getpagesize` is pretty safe, It returns a constant value. 

https://android.googlesource.com/platform/bionic/+/master/libc/platform/bionic/page.h#26

```fish
❯ lldb -- ./a.out
(lldb) target create "./a.out"
Current executable set to '/data/data/com.termux/files/home/a.out' (aarch64).
(lldb) b getpagesize
Breakpoint 1: no locations (pending).
WARNING:  Unable to resolve breakpoint to any actual locations.
(lldb) run
Process 18654 launched: '/data/data/com.termux/files/home/a.out' (aarch64)
1 location added to breakpoint 1
Process 18654 stopped
* thread #1, name = 'a.out', stop reason = breakpoint 1.1
    frame #0: 0x0000007fb7cc8458 libc.so`getpagesize
libc.so`getpagesize:
->  0x7fb7cc8458 <+0>: orr    w0, wzr, #0x1000
    0x7fb7cc845c <+4>: ret    

libc.so`getpgrp:
    0x7fb7cc8460 <+0>: mov    w0, wzr
    0x7fb7cc8464 <+4>: b      0x7fb7cbf690   ; symbol stub for: getpgid
(lldb) dis
libc.so`getpagesize:
->  0x7fb7cc8458 <+0>: orr    w0, wzr, #0x1000
    0x7fb7cc845c <+4>: ret    

```

Even though it's deprecated, Android still recommends it.
https://source.android.com/docs/core/architecture/16kb-page-size/getting-page-size#use-code